rmw_zenoh_cpp/src/detail/attachment_helpers.cpp

```diff
@@ -122,17 +73,38 @@ size_t AttachmentData::gid_hash() const
   return gid_hash_;
 }
 
-///=============================================================================
-void AttachmentData::serialize_to_zbytes(z_owned_bytes_t * attachment)
+zenoh::Bytes AttachmentData::serialize_to_zbytes()
+{
+  auto serializer = zenoh::ext::Serializer();
+  serializer.serialize(std::string("sequence_number"));
+  serializer.serialize(this->sequence_number_);
+  serializer.serialize(std::string("source_timestamp"));
+  serializer.serialize(this->source_timestamp_);
+  serializer.serialize(std::string("source_gid"));
+  serializer.serialize(this->source_gid_);
+  return std::move(serializer).finish();
+}
+
+AttachmentData::AttachmentData(const zenoh::Bytes & bytes)
 {
-  ze_owned_serializer_t serializer;
-  ze_serializer_empty(&serializer);
-  ze_serializer_serialize_str(z_loan_mut(serializer), "sequence_number");
-  ze_serializer_serialize_int64(z_loan_mut(serializer), this->sequence_number_);
-  ze_serializer_serialize_str(z_loan_mut(serializer), "source_timestamp");
-  ze_serializer_serialize_int64(z_loan_mut(serializer), this->source_timestamp_);
-  ze_serializer_serialize_str(z_loan_mut(serializer), "source_gid");
-  ze_serializer_serialize_buf(z_loan_mut(serializer), this->source_gid_, RMW_GID_STORAGE_SIZE);
-  ze_serializer_finish(z_move(serializer), attachment);
+  zenoh::ext::Deserializer deserializer(std::move(bytes));
```

Comment:
```suggestion
  zenoh::ext::Deserializer deserializer(bytes);
```

(we don't want to take ownership of these bytes)